doc/whats_new/v1.2.rst

```diff
@@ -496,6 +496,17 @@ Changelog
   :class:`neighbors.RadiusNeighborsTransformer`.
   :pr:`24075` by :user:`Valentin Laurent <Valentin-Laurent>`.
 
+:mod:`sklearn.neural_network`
+.............................
+
+- |Fix| :class:`neural_network.MLPClassifier` and
+  :class:`neural_network.MLPRegressor` always expose the parameters `best_loss_`,
+  `validation_scores_`, and `best_validation_score_`. `best_loss_` is set to
+  `None` when `early_stopping=True`, while `validation_scores_` and
+  `best_validation_score_` are set to `None` when `early_stopping=False`.
+  `validation_scores_` also changed from a Python list to a Numpy array.
```

Comment:
I do not think we can make `validation_scores_` a ndarray because it breaks `warm_start`:

```python
from sklearn.neural_network import MLPRegressor
from sklearn.datasets import make_regression

X, y = make_regression(random_state=42)
mlp = MLPRegressor(max_iter=10, random_state=0, warm_start=True, early_stopping=True)

mlp.fit(X, y)

mlp.set_params(max_iter=20)

# Fails with this PR
mlp.fit(X, y)
```

Reply:
Good catch. I will revert this and add a non-regression test as well.